generator/src/main/java/com/scottlogic/deg/generator/inputs/IndividualConstraintRuleViolator.java

```diff
@@ -0,0 +1,140 @@
+        //   OR(
+        //     AND(VIOLATE(X), Y, Z),
+        //     AND(X, VIOLATE(Y), Z),
+        //     AND(X, Y, VIOLATE(Z)))
+        if (constraint instanceof AndConstraint) {
+            Collection<Constraint> subConstraints = ((AndConstraint) constraint).subConstraints;
+
+            Collection<Constraint> violatedIndividually =
+                subConstraints.stream()
+                    // for each subconstraint X, make a copy of the original list but with X replaced by VIOLATE(X)
+                    .map(constraintToViolate ->
+                        subConstraints.stream()
+                            .map(c -> c == constraintToViolate
+                                ? violateConstraint(c)
+                                : c)
+                            .collect(Collectors.toList()))
+                    // make an AndConstraint out of each of the new lists
+                    .map(AndConstraint::new)
+                    .collect(Collectors.toList());
+
+            return new OrConstraint(violatedIndividually);
+        }
+        // VIOLATE(OR(X, Y, Z)) reduces to AND(VIOLATE(X), VIOLATE(Y), VIOLATE(Z))
+        else if (constraint instanceof OrConstraint) {
+            Collection<Constraint> subConstraints = ((OrConstraint) constraint).subConstraints;
+
+            Collection<Constraint> violatedAll =
+            subConstraints.stream()
+                .map(this::violateConstraint)
+                .collect(Collectors.toList());
+
+            return new AndConstraint(violatedAll);
+        }
+        // VIOLATE(IF(X, then: Y)) reduces to AND(X, VIOLATE(Y))
+        // VIOLATE(IF(X, then: Y, else: Z)) reduces to OR(AND(X, VIOLATE(Y)), AND(VIOLATE(X), VIOLATE(Z)))
+        else if (constraint instanceof ConditionalConstraint) {
+            ConditionalConstraint conditional = ((ConditionalConstraint) constraint);
+
+            Constraint positiveViolation = new AndConstraint(
+                conditional.condition,
+                violateConstraint(conditional.whenConditionIsTrue));
+
+            Constraint negativeViolation = conditional.whenConditionIsFalse == null
+                ? null
+                : new AndConstraint(
+                violateConstraint(conditional.condition),
+                violateConstraint(conditional.whenConditionIsFalse));
+
+            return  negativeViolation != null
+                    ? new OrConstraint(positiveViolation, negativeViolation)
+                    : positiveViolation;
+        }
+        // VIOLATE(AtomicConstraint) reduces to NEGATE(AtomicConstraint)
+        // We wrap this in a ViolatedAtomicConstraint to allow Visualise to show which constraint is being violated
+        else if (constraint instanceof AtomicConstraint) {
+            if (canViolate(constraint)) {
+                return new ViolatedAtomicConstraint((AtomicConstraint)constraint.negate());
+            }
+            return constraint;
+        }
+        else {
```

Comment:
This is missing  

        (constraint instanceof NegatedGramaticalConstraint)

Reply:
@pdaulbyscottlogic please re-review